src/servers/Server.cc

```diff
@@ -146,6 +146,9 @@ Server::doClientRead(const CommIoCbParams &io)
 
     CommIoCbParams rd(this); // will be expanded with ReadNow results
     rd.conn = io.conn;
+    Assure(Config.maxRequestBufferSize > inBuf.length());
+    rd.size = Config.maxRequestBufferSize - inBuf.length();
```

Comment:
> Please be aware that `maybeMakeSpaceAvailable()` already restricts capacity to `Config.maxRequestBufferSize`.

maybeMakeSpaceAvailable() does not restrict capacity (enough): The method may increase capacity up to the specified maximum (if needed), but it does not _shrink_ already excessive capacity.

> when this buffer contains `maxRequestBufferSize` bytes, the behaviour you are trying to avoid **will** occur despite this change.

Existing mayBufferMoreRequestBytes() check executed a few lines earlier makes the precondition in the above statement false -- this buffer cannot contain `maxRequestBufferSize` bytes at this point. That check was added in recent master commit 4d6dd3dd to fix another bug. That is why this PR can assert that maxRequestBufferSize exceeds inBuf.length() and, hence, this rd.size math works without underflows.